cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/validator/AwsIDBrokerAssumeRoleValidator.java

```diff
@@ -48,8 +48,12 @@ public boolean canAssumeRoles(AmazonIdentityManagementClient iam, InstanceProfil
             } catch (AmazonIdentityManagementException e) {
                 // Log the error and return true. We don't want to block if there is an IAM failure.
                 // This can happen due to throttling or other issues.
+                // If error messages access denied we add the error to the result
                 LOGGER.error("Unable to check assume role from instance profile {} for roles {} due to {}",
                         instanceProfile.getArn(), roleArns, e.getMessage(), e);
+                if ("AccessDenied".equals(e.getErrorCode())) {
+                    resultBuilder.error("Unable to check assume role: " + e.getErrorMessage());
```

Comment:
If there is an error I think it should return false

Reply:
We have never use the return value, so I remove the return value

Reply:
We have never use the return value, so nevermind. I will return with false